src/policy/compiler.rs

```diff
@@ -164,38 +169,64 @@ impl Property for CompilerExtData {
             branch_prob: None,
             sat_cost: f64::MAX,
             dissat_cost: Some(0.0),
+            sig_type: PhantomData,
         }
     }
 
     fn from_pk_k() -> Self {
         CompilerExtData {
             branch_prob: None,
-            sat_cost: 73.0,
+            sat_cost: match Ctx::sig_type() {
+                SigType::Ecdsa => 73.,
+                SigType::Schnorr => 64.,
```

Comment:
I think this should probably be 66 so that we include the `varint` prefix and the hash type flag. This is because the key itself is `64` bytes. It can optionally contain a sighash flag (1 byte) and there is also a pre-fix of `varint` type denoting the length of sig that follows. It will end up as  `<var_int_prefix><sig><sighash_type>` on the stack.

We should also add a comment explaining why this is 66 and we need to update this everywhere.

Reply:
Addressed in 2565f5e